byron/semantics/executable-spec/src/Control/State/Transition.hs

```diff
@@ -52,6 +52,12 @@ newtype IRC sts = IRC (Environment sts)
 -- | Context available to transition rules.
 newtype TRC sts = TRC (Environment sts, State sts, Signal sts)
 
+deriving instance
```

Comment:
I believe you could just write `deriving (Show)` on the `newtype` declaration

Reply:
This would impose the Show constraint on `Environment sts`, `State sts` and `Signal sts`, which isn't there at the moment. This way it's flexible and not imposed globally.